pyxrf/model/draw_image_rgb.py

```diff
@@ -0,0 +1,353 @@
+        if len(scaler_groups) > 0:
+            #self.scaler_group_name = scaler_groups[0]
+            self.scaler_norm_dict = self.data_dict[scaler_groups[0]]
+            # for GUI purpose only
+            self.scaler_items = []
+            self.scaler_items = self.scaler_norm_dict.keys()
+            self.scaler_data = None
+
+        self.show_image()
+
+    @observe('data_opt')
+    def _update_file(self, change):
+        if self.data_opt == 0:
+            self.dict_to_plot = {}
+            self.items_in_selected_group = self.dict_to_plot.keys()
+            self.set_stat_for_all(bool_val=False)
+        elif self.data_opt > 0:
+            self.set_stat_for_all(bool_val=False)
+            plot_item = sorted(self.data_dict_keys)[self.data_opt-1]
+            self.dict_to_plot = self.data_dict[plot_item]
+            # for GUI purpose only
+            self.items_in_selected_group = []
+            self.items_in_selected_group = self.dict_to_plot.keys()
+            self.set_stat_for_all(bool_val=False)
+
+    @observe('scaler_name_index')
+    def _get_scaler_data(self, change):
+        if self.scaler_name_index == 0:
+            self.scaler_data = None
+        else:
+            scaler_name = self.scaler_items[self.scaler_name_index-1]
+            #self.scaler_data = self.data_dict[self.scaler_group_name][scaler_name]
+            self.scaler_data = self.scaler_norm_dict[scaler_name]
+            logger.info('Use scaler data to normalize,'
+                        'and the shape of scaler data is {}'.format(self.scaler_data.shape))
+        self.show_image()
+
+    @observe('scale_opt')
+    def _update_scale(self, change):
+        if change['type'] != 'create':
+            self.show_image()
+
+    def set_stat_for_all(self, bool_val=False):
+        """
+        Set plotting status for all the 2D images.
+        """
+        self.stat_dict.clear()
+        self.stat_dict = {k: bool_val for k in self.items_in_selected_group}
+
+    def preprocess_data(self):
+        """
+        Normalize data or prepare for linear/log plot.
+        """
+
+        selected_data = []
+        selected_name = []
+
+        stat_temp = self.get_activated_num()
+        stat_temp = OrderedDict(sorted(six.iteritems(stat_temp), key=lambda x: x[0]))
+
+        low_lim = 1e-4  # define the low limit for log image
+        ic_norm = 10000  # multiply by this value for ic normalization
+        plot_interp = 'Nearest'
+        name_not_scalable = ['r_squared']  # do not apply scaler norm on those data
+
+        if self.scaler_data is not None:
+            if np.max(self.scaler_data) < 1.0:  # use current as ic, such as SRX
+                ic_norm = 1.0
+            if len(self.scaler_data[self.scaler_data == 0]) > 0:
+                logger.warning('scaler data has zero values at {}'.format(np.where(self.scaler_data == 0)))
+                self.scaler_data[self.scaler_data == 0] = np.mean(self.scaler_data)
```

Comment:
Also, you probably want to take the mean of the non-zero values?
`self.scaler_data[self.scaler_data == 0] = np.mean(self.scaler_data[self.scaler_data != 0])`